Timezone support

ramesius opened the pull request:
Fixes #21

ramesius commented:
Merge after #17

ramesius commented on events.js:
```diff
@@ -0,0 +1,48 @@
+const operatingTimezones = require('./operatingTimezones');
+const isEnabled = require('./isEnabled').isEnabled;
+
+const UTC_START_HOUR = 9;
```
These constants now control the start and stop hour of hammertime

ramesius commented on operatingTimezones.js:
```diff
@@ -0,0 +1,5 @@
+const timezones = process.env.HAMMERTIME_OPERATING_TIMEZONES || '[10]';
+
+const parsed = JSON.parse(timezones);
```
Timezones should be express as a javascript array in the environment variable.

ramesius commented on src/tags/operatingTimezone.js:
```diff
@@ -0,0 +1,19 @@
+
+  if (timezoneTags.length === 0 || timezoneTags.length > 1) {
+    console.log('No timezone tag found or multiple timezone tags found. Defaulting to +10');
+    return 10;
```
Default to AEDT (+10) for resources that do not have a `hammertime:operatingTimezone` tag

krutisfood replied:
Config file would be ideal but the console log is nice and clear, the filename seems obvious enough, nice!

ramesius commented on events.js:
```diff
@@ -0,0 +1,48 @@
+}
+
+module.exports.stop = stop;
+module.exports.start = start;
```
These functions generate the schedule events that end up being put into the serverless.yml on deployment. It works similar to how the dryRun and isEnabled environment variable javascript files work.

krutisfood commented on test/asgs/listASGsToStart.test.js:
```diff
@@ -9,7 +9,7 @@ describe('listASGsToStart()', () => {
     AWS.mock('AutoScaling', 'describeAutoScalingGroups', startOnePageResponse);
 
-    return listASGsToStart()
+    return listASGsToStart(10)
```
This would be more obvious as a variable called "defaultTimeZone", thoughts?

ramesius replied:
Indeed, will sort that out.

krutisfood commented:
needs a version bump pls.

krutisfood commented:
Can we change this PR to master now?
Also needs some updates to the README so we know how it works :)
Early testing is looking good though, we've got multiple cloudwatch events with two timezones configured 👍

ramesius commented:
Yeah I think a nice write up in the readme will definetly be worth it. Also after some thinking I would like to see if I can include configuration for timezones using the format 'Australia/Sydney' so DST can be taken into account when deciding to start/stop an asset.

ramesius commented:
`m().tz('Australia/Sydney').utcOffset()` I think this will suffice getting the timezone offeset, will implement this soon.

ramesius commented:
Have fixed up a few rough edges in the code. Switched out moment for luxon which provided a nicer API for dealing with the date and time stuff. Added the ability to configure the start and stop time of hammertime. Have fixed up some things in the readme also.

krutisfood commented:
For some reason hasn't picked up any of the ASGs, though happily spun down our instances
```11:01:24 ... No ASGs to spin down, moving on...```

krutisfood commented on .travis.yml:
```diff
@@ -1,5 +1,8 @@
 language: node_js
 node_js:
-  - "node"
+  - "8"
```
Bit late to the party, but should this reflect the version in serverless.yml? or are we just anticipating the next PR :)

krutisfood commented on serverless.yml:
```diff
@@ -33,20 +33,13 @@ provider:
 functions:
   stop-hammertime:
     handler: hammertime.stop
     timeout: 180
```
We need a longer timeout these days, could be subsequent PR though to make it a parameter :)

krutisfood commented:
Spun down 316 ASGs and half dozen instances, nice! We'll see how well it boots everything up in the morning & must be time to put this PR to bed!
